Crash - Timer already canceled

If you swap out to another app and back again while on either the Pre- or Match activity, you'll get:
(we need to check something before trying to cancel the timer.

 FATAL EXCEPTION: main
                                                                                                    Process: com.team3663.scouting_app, PID: 5177
                                                                                                    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.team3663.scouting_app/com.team3663.scouting_app.activities.AppLaunch}: java.lang.IllegalStateException: Timer already cancelled.
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3782)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3922)
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139)
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96)
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443)
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106)
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205)
                                                                                                    	at android.os.Looper.loop(Looper.java:294)
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8177)
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method)
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552)
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971)
                                                                                                    Caused by: java.lang.IllegalStateException: Timer already cancelled.
                                                                                                    	at java.util.Timer.sched(Timer.java:405)
                                                                                                    	at java.util.Timer.schedule(Timer.java:201)
                                                                                                    	at com.team3663.scouting_app.activities.AppLaunch.loadData(AppLaunch.java:163)
                                                                                                    	at com.team3663.scouting_app.activities.AppLaunch.onCreate(AppLaunch.java:121)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8595)
                                                                                                    	at android.app.Activity.performCreate(Activity.java:8573)
                                                                                                    	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1456)
                                                                                                    	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3764)
                                                                                                    	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3922) 
                                                                                                    	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:103) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:139) 
                                                                                                    	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:96) 
                                                                                                    	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2443) 
                                                                                                    	at android.os.Handler.dispatchMessage(Handler.java:106) 
                                                                                                    	at android.os.Looper.loopOnce(Looper.java:205) 
                                                                                                    	at android.os.Looper.loop(Looper.java:294) 
                                                                                                    	at android.app.ActivityThread.main(ActivityThread.java:8177) 
                                                                                                    	at java.lang.reflect.Method.invoke(Native Method) 
                                                                                                    	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:552) 
                                                                                                    	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:971) 

--- a/app/src/main/java/com/team3663/scouting_app/activities/AppLaunch.java
+++ b/app/src/main/java/com/team3663/scouting_app/activities/AppLaunch.java
@@ -103,7 +103,8 @@ public class AppLaunch extends AppCompatActivity {
         appLaunchBinding.textVersion.setText(getString(R.string.app_version) + " " + pInfo.versionName);
 
         // Get the Shared Preferences where we save off app settings to use next time
-        if (Globals.sp == null) Globals.sp = this.getSharedPreferences(getString(R.string.preference_setting_file_key), Context.MODE_PRIVATE);
+        if (Globals.sp == null)
+            Globals.sp = this.getSharedPreferences(getString(R.string.preference_setting_file_key), Context.MODE_PRIVATE);
         if (Globals.spe == null) Globals.spe = Globals.sp.edit();
 
         // Initialize activity components
@@ -119,8 +120,7 @@ public class AppLaunch extends AppCompatActivity {
         if (havePerms) {
             Globals.baseStorageURI = Uri.parse(Globals.sp.getString(Constants.Prefs.STORAGE_URI, null));
             loadData();
-        }
-        else
+        } else
             initStoragePermissions();
 
         // While loading Matches, we messed with Globals.CurrentMatchType, so reset it
@@ -133,8 +133,7 @@ public class AppLaunch extends AppCompatActivity {
     // Output:      void
     // Parameters:  void
     // =============================================================================================
-    private void loadData()
-    {
+    private void loadData() {
         // Make sure we have our directory structure created
         DocumentFile storage_df = DocumentFile.fromTreeUri(this, Globals.baseStorageURI);
 
@@ -294,7 +293,7 @@ public class AppLaunch extends AppCompatActivity {
 
         // Ensure the public file exists, and if not, copy the private one there.
         // Return back if we should use the private or public file.
-       usePublic = CopyPrivateToPublicFile(in_fileName, in_msgError);
+        usePublic = CopyPrivateToPublicFile(in_fileName, in_msgError);
 
         // Update the loading status
         appLaunchBinding.textStatus.setText(in_msgLoading);
@@ -401,8 +400,10 @@ public class AppLaunch extends AppCompatActivity {
         appLaunchBinding.butStartScouting.setClickable(false);
         appLaunchBinding.butStartScouting.setOnClickListener(view -> {
             // Stop the timer
-            appLaunch_timer.cancel();
-            appLaunch_timer.purge();
+            if (appLaunch_timer != null) {
+                appLaunch_timer.cancel();
+                appLaunch_timer.purge();
+            }
 
             // Default Globals
             Globals.CurrentTeamOverrideNum = 0;
@@ -410,7 +411,7 @@ public class AppLaunch extends AppCompatActivity {
             if ((Globals.sp == null) ||
                     (Globals.sp.getInt(Constants.Prefs.COMPETITION_ID, -1) == -1) ||
                     (Globals.sp.getInt(Constants.Prefs.DEVICE_ID, -1) == -1)) {
-                Toast.makeText( AppLaunch.this,R.string.applaunch_not_configured, Toast.LENGTH_SHORT).show();
+                Toast.makeText(AppLaunch.this, R.string.applaunch_not_configured, Toast.LENGTH_SHORT).show();
             } else {
                 // Go to the first page
                 Intent GoToPreMatch = new Intent(AppLaunch.this, PreMatch.class);

